test/unit/query-generator/drop-table-query.test.ts

```diff
@@ -28,8 +29,14 @@ describe('QueryGenerator#dropTableQuery', () => {
     });
   });
 
-  // FIXME: enable this test once fixed (in https://github.com/sequelize/sequelize/pull/14687)
-  it.skip('produces a DROP TABLE query from a table and globally set schema', () => {
+  it('produces a DROP TABLE query with default schema', () => {
```

Comment:
title should reflect that it does not include the schema in the query if the schema is the default schema

---

I remain very unsure about this change. I have two conflicting views:

1. the schema should always be specified to ensure stability
2. we don't want to break the use case where someone sets the 'searchPath' option to target different tables using the same model.

I think I may have been mistaken in my approach in the previous PR. I think point 1 is more important. I think point 2 should not be handled using searchPath but by supporting a "schema" option in most model methods, instead of "searchPath".

That's still allow users to use the same model to create multiple tables, while being much more predictable.

Like this:

```ts
class User extends Model {}
User.init(/* ... */);

User.sync(); // synched in default schema
User.sync({ schema: 'client1' }); // synched in client1 schema (this is already possible)
User.findAll() // find from default schema
User.findAll({ schema: 'client1' }) // find from "client1" schema (not currently possible, users use searchPath instead)
```

What do you think? Let's make `quoteTable` *always* specify the schema even if it's the default schema?

Reply:
That would be the safer option, but let's do that in a separate PR.

I'll fix it in here for the dialects that don't use `quoteTable`. About that, maybe the default schema for DB2 is `USER` and IBMi is `CURRENT SCHEMA`?

Reply:
We can't use `CURRENT SCHEMA` as part of an identifier, `getDefaultSchema` needs to return the actual schema name